domain-models-runtime/src/main/resources/templates/db_scripts/general_functions.ftl

```diff
@@ -67,18 +67,20 @@ $$ LANGUAGE sql IMMUTABLE PARALLEL SAFE STRICT;
 
 -- Normalize digits by removing spaces, tabs and hyphen-minuses from the first chunk.
 -- Insert a space before the second chunk. The second chunk starts at the first character that is
--- neither digit, space, tab nor hyphen-minus.
+-- neither digit, space, tab nor hyphen-minus. The first chunk may end with a star * for right
+-- truncation.
 -- Examples:
 -- normalize_digits(' 0-1  2--3 4 ') = '01234'
 -- normalize_digits(' 01 2- 3 -- 45 -a 7 -8 9') = '012345 a 7 -8 9'
+-- normalize_digits(' 01 2- 3 -- 45* -a 7 -8 9') = '012345* a 7 -8 9'
 -- normalize_digits('978 92 8011 565 9(Vol. 1011-1021)') = '9789280115659 (Vol. 1011-1021)'
 CREATE OR REPLACE FUNCTION ${myuniversity}_${mymodule}.normalize_digits(text) RETURNS text AS $$
-  SELECT    translate((regexp_match($1, '^([0-9 \t-]*)(.*)'))[1], E' \t-', '')
-         || CASE WHEN (regexp_match($1, '^([0-9 \t-]*)(.*)'))[1] = '' THEN ''
-                 WHEN (regexp_match($1, '^([0-9 \t-]*)(.*)'))[2] = '' THEN ''
+  SELECT    translate((regexp_match($1, '^([0-9 \t-]*(?:\*[ \t]*)?)(.*)'))[1], E' \t-', '')
+         || CASE WHEN (regexp_match($1, '^([0-9 \t-]*(?:\*[ \t]*)?)(.*)'))[1] = '' THEN ''
+                 WHEN (regexp_match($1, '^([0-9 \t-]*(?:\*[ \t]*)?)(.*)'))[2] = '' THEN ''
                  ELSE ' '
```

Comment:
You are redefining a function. If not replaced properly, the old definition sticks. Therefore, rename it (and the old definition will stay for older systems (eg 29.3.0 RMB), but will never be used).

Reply:
Can you explain in which cases "CREATE OR REPLACE FUNCTION" may not replace the old definition?

Reply:
I can't. Would error out if parameters or other would change. Will indexes based on updating the function just be updated? Or is that a mood question because AFAIK we are always regenerating indices anyway?

Reply:
The change allows a star at the end of the barcode. This star is punctuation that the full text index removes and ignores. Therefore no index regenerating is needed for this change.
This star affects the query string only where it signals right truncation.

Reply:
ok